packages/basti/src/aws/elasticache/get-elasticache-memcached-clusters.ts

```diff
@@ -0,0 +1,90 @@
+import { DescribeCacheClustersCommand } from '@aws-sdk/client-elasticache';
+
+import { AwsNotFoundError } from '../common/aws-errors.js';
+
+import {
+  parseMemcachedCacheClusteWithSpecificNoderResponse,
+  parseMemcachedCacheClusterResponse,
+} from './parse-elasticache-memcached-response.js';
+import { elasticacheClient } from './elasticache-client.js';
+
+import type { CacheCluster } from '@aws-sdk/client-elasticache';
+import type { AwsElasticacheMemcachedCluster } from './elasticache-types.js';
+
+export interface getCacheClusterInput {
+  identifier: string;
+}
+
+export async function getMemcachedClusters(): Promise<
+  AwsElasticacheMemcachedCluster[]
+> {
+  const { CacheClusters } = await elasticacheClient.send(
+    new DescribeCacheClustersCommand({
+      ShowCacheNodeInfo: true,
+      ShowCacheClustersNotInReplicationGroups: true,
+    })
+  );
+
+  if (!CacheClusters) {
+    throw new Error(`Invalid response from AWS.`);
+  }
+  const memcachedClusters = await getRawMemcachedClusters();
+  return memcachedClusters.map(cacheCluster => {
+    return parseMemcachedCacheClusterResponse(cacheCluster);
+  });
+}
+
+export async function getRawMemcachedClusters(): Promise<CacheCluster[]> {
+  const { CacheClusters } = await elasticacheClient.send(
+    new DescribeCacheClustersCommand({
+      ShowCacheNodeInfo: true,
+      ShowCacheClustersNotInReplicationGroups: true,
+    })
+  );
+
+  if (!CacheClusters) {
+    throw new Error(`Invalid response from AWS.`);
+  }
+  const memcachedClusters = CacheClusters.filter(
+    cacheCluster => cacheCluster.Engine === 'memcached'
+  );
+  return memcachedClusters;
+}
+
+export async function getMemcachedCluster(
+  identifier: string
+): Promise<AwsElasticacheMemcachedCluster> {
+  if (identifier === undefined) {
```

Comment:
Could you please, explain why we need these checks before removing them? I mean, we need to fix the root cause